packages/searchinput/src/react/index.tsx

```diff
@@ -1,40 +1,48 @@
 import { css } from 'glamor'
-import React, { useRef, useState } from 'react'
-import PropTypes from 'prop-types'
+import React, { useRef, useState, HTMLAttributes } from 'react'
 
 import Button from '@pluralsight/ps-design-system-button'
 import CircularProgress from '@pluralsight/ps-design-system-circularprogress'
 import { CloseIcon, SearchIcon } from '@pluralsight/ps-design-system-icon'
 import TextInput from '@pluralsight/ps-design-system-textinput'
 import { callAll } from '@pluralsight/ps-design-system-util'
 
-import stylesheet from '../css/index.js'
+import stylesheet from '../css'
 
 const styles = {
-  clear: clearVisible =>
+  clear: (clearVisible: boolean) =>
     css(
       stylesheet['.psds-searchinput-clear'],
       clearVisible && stylesheet['.psds-searchinput-clear__visible']
     ),
-  field: _ => css(stylesheet['.psds-searchinput-field'])
+  field: () => css(stylesheet['.psds-searchinput-field'])
 }
 
-const SearchInput = React.forwardRef(
-  ({ loading, onClear, onChange, ...rest }, forwardedRef) => {
+export interface SearchInputProps
+  extends Omit<HTMLAttributes<HTMLInputElement>, 'onChange'> {
+  loading?: boolean
+  onClear?: (evt?: React.MouseEvent) => void
```

Comment:
I made that optional because as a developer I can see myself enabling clearing but not caring about the evt. Thus if I'm using typescript I don't want to force a user to pass a callback function that has to take an argument. A noop is just as good like in the original story the team wrote. I could be wrong but I thought unless that was made optional then noop would throw an error in typescript.

Reply:
Yeah, ok, I can see that. I think that does seem nice.  We should probably chat about that as a team to make sure we're all on the same page on the approach and the reasons for it.  (I just saw a commit a few days ago that was changing an already-converted TS package from having optional events in an event handler to be required.)

cc @danethurber , looping you in.  Please take a look at the code and comments and chime in with how you see this issue.